### DISCUSSION TITLE
NS3 log file too big

### DISCUSSION THREAD

**Discussion by DFCui** [eclipse/mosaic#153] [Q&A] [answered]
Hello all,
I run a scenario with NS3 enabled and found the log file(communication.log) for NS3 is big. I checked the log file and found that most log information is for process  org.eclipse.mosaic.fed.ns3.ambassador.Ns3Ambassador, i.e. Ns3AmbassadorOutput, which are details about NS3 simulation. I am not interested these details and they take too much space. 
Is there anyone knowing how to turn off the logger for the process org.eclipse.mosaic.fed.ns3.ambassador.Ns3Ambassador?
Could you please provide some suggestions on how to disable the Ns3AmbassadorOutput?

Thank you very much.

**Comment by kschrab** [comment, marked as answer]
Hi, the main configuration for logging in MOSAIC is in `etc/logback.xml`. To disable the log for Ns3Ambassador completely, you can change the log level for the logger named `"org.eclipse.mosaic.fed.ns"` to `WARN` or `ERROR`:
```xml
<logger name="org.eclipse.mosaic.fed.ns3" additivity="false" level="ERROR">
    <appender-ref ref="CommunicationLog"/>
</logger>
```

To configure the logging of the ns-3 federate, than you need to configure the logging in the top section of the `ns3_federate_config.xml` of the scenario. Here we already pre-configured the logging level for each component of ns-3, to make it simpler for the user. Make sure to use the latest version of MOSAIC , as we just improved the logging a few days ago (use 21.1 release from today, or build from the current main branch).
